src/main/java/frc/robot/commands/drive/Drive.java

```diff
@@ -26,6 +23,34 @@ public Drive() {
         requires(Drivetrain.getInstance());
     }
 
+    public static double powerCurve(double x) {
+        // https://www.desmos.com/calculator/g07ukjj7bl
+        double curve = (0.5D * (Math.atan(Math.PI * (Math.abs(x) - 0.5D)))) + 0.5D;
+        return Math.copySign(curve, x);
+    }
+
+    /**@param array array of motor power values
+     * If any of the values are more than 1, they aren't valid values for motor power.
+     * If so, it divides all array values by the largest value to preserve the value ratios while making them valid motor power values.
+    */
+    public static void normalize(double[] array) {
```

Comment:
Better placement, and thank you for the documentation. I still feel less-than-happy with the parameter name, but I wouldn't block merge for that alone (other, stricter people certainly would though, because it's painfully non descriptive.)

Reply:
Also, I'll fix this myself later, but please read up on how to write [Javadoc.](https://en.wikipedia.org/wiki/Javadoc#Overview_of_Javadoc)